source/puppet/4.1/config_ssl_external_ca.markdown

```diff
@@ -201,230 +107,70 @@ Credential                        | File location
+* The CA certificate bundle (ie the external Root CA combined with the Intermediate CA certificate) must be distributed to the agents manually - ideally before puppet runs
 
-{{ master_basic }}
+### Puppet master
 
-You must also create a **CA bundle** for the web server. Append the **Agent CA certificate and Root CA certificate** together; the Root CA certificate must be located after the Agent CA certificate within the file.
+Before configuring the puppet master, you will need to obtain the intermediate CA certificate from your external Root CA.  Generating the Intermediate CA cert is outside the scope of the doc, since it will depend your external certificate authority solution.  This guide assumes you are either starting with a fresh installation or have removed all SSL files from your existing master and are starting over.  Also, you should stop all puppet related services on the master server before this process.
 
-~~~ bash
-cat agent_ca.pem root_ca.pem > ca_bundle_for_master.pem
-~~~
+In order to configure the puppet master you will need the following files placed:
 
-Put this file somewhere predictable. Puppet doesn't use it directly, but it can live alongside Puppet's copies of the certificates.
+Certificate     | Purpose                     | File Location
+----------------|--------------------------------------------
+ca_crt.pem      | Intermediate CA Certificate | /etc/puppetlabs/puppet/ssl/ca/ca_crt.pem
+ca_key.pem      | Intermediate CA Key         | /etc/puppetlabs/puppet/ssl/ca/ca_key.pem
+root_crt.pem    | Root CA Certificate         | /etc/puppetlabs/puppet/ssl/ca/root_crt.pem
 
-With these files in place, the web server should be configured to:
+> Note: The root_crt.pem can actually be placed anywhere, however this doc assumes you placed it as shown
 
-* Use the Agent+Root CA bundle, the master's certificate, and the master's key.
-* Set the verification header (as specified in the master's [`ssl_client_verify_header` setting][verify_header]).
-* Set the client DN header (as specified in the master's [`ssl_client_header` setting][client_header]).
+> Note: The ca_key.pem needs to have any passphrase removed to match the expectations of the Puppet CA
 
-An example of this configuration for Apache:
+All of the files placed should have owner set to `puppet:puppet` and permissions of `0600`.
 
-~~~ apache
-Listen 8140
-<VirtualHost *:8140>
-    SSLEngine on
-    SSLProtocol ALL -SSLv2
-    SSLCipherSuite ALL:!ADH:RC4+RSA:+HIGH:+MEDIUM:-LOW:-SSLv2:-EXP
+Next, you have to generate the CA bundle to be placed on the master as well as any agents you create.  This is achieved by combining the Root CA and Intermediate CA certificates into one PEM file.
 
-    # Replace with the value of `puppet master --configprint hostcert`
-    SSLCertificateFile "/path/to/master.pem"
-    # Replace with the value of `puppet master --configprint hostprivkey`
-    SSLCertificateKeyFile "/path/to/master.key"
+```
+cd /etc/puppetlabs/puppet/ssl/ca
```

Comment:
See the earlier comments about conflating `$ssldir` (`/etc/puppetlabs/puppet/ssl`) and `$cadir` (`/etc/puppetlabs/puppet/ssl/ca`).